src/applause-button.scss

```diff
@@ -1,145 +1,176 @@
-@mixin animate-once($name) {
-  animation-name: $name;
-  animation-duration: 0.5s;
-  animation-iteration-count: 1;
-}
+applause-button {
+  --disabled-color: #707070;
+  --animation-duration: 0.5s;
+  --button-border-width: 0.125rem;
+
+  box-sizing: border-box;
+  display: block;
+  scale: 1;
+  /*
+    Backwards compatibility: previously width and height needed to be set,
+    but now we lock 1:1 aspect ratio and use width to determine size,
+    so we must override/reset any user-specified height
+  */
+  height: initial !important;
 
-@mixin centered-square($size) {
-  position: absolute;
-  width: $size;
-  height: $size;
-  margin-left: calc((100% - $size) / 2);
-  margin-top: calc((100% - $size) / 2);
-}
+  &.loading {
+    opacity: 0.5;
+  }
 
-@mixin position-fill() {
-  position: absolute;
-  top: 0;
-  left: 0;
-  right: 0;
-  bottom: 0;
-}
+  /* CSS that is toggled to fire clap animation */
+  &.clap {
+    animation-name: pulse;
+    animation-duration: var(--animation-duration);
 
-applause-button {
-  position: relative;
-  cursor: pointer;
+    .clap-count {
+      animation-name: hideThenShow;
+      animation-duration: var(--animation-duration);
+    }
+    svg g.sparkle circle {
+      animation-name: starPulse;
+      animation-duration: var(--animation-duration);
+    }
+  }
 
-  .style-root {
-    @include position-fill();
+  &.clap-limit-exceeded div {
+    fill: var(--disabled-color) !important;
```

Comment:
Is the `!important` important here (so to speak 😆)?

Reply:
Yep! We allow users to set colour via `color` attribute on our custom element, and as a result that sets color properties as `style` on the style-root div. But because that is inline style, the only way to override that (which we need to do when we want "button is now disabled" styling) is to use the `!important` escape hatch.

I wonder if we could instead set a CSS property as a result of that attribute change, then we wouldn't need this `!important` which does feel a bit ick. I'll try it out.

Incidentally, `attributeChangedCallback` is a built-in method of custom elements, invoked whenever an attribute is, uh, changed. Our custom element listens for the `color` change - see also `observedAttributes` property.

Reply:
Well that worked nicely, and allowed a little code cleanup ✨